plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3339,4 +3340,43 @@ public void testOptimizeParameterValidation()
                 "ALTER TABLE nation EXECUTE OPTIMIZE (file_size_threshold => '33s')",
                 "\\QUnable to set procedure property 'file_size_threshold' to ['33s']: Unknown unit: s");
     }
+
+    @Test
+    public void testRepartitioningIsForcedDuringOptimize()
+            throws IOException
+    {
+        String tableName = "test_repartitiong_during_optimize_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (key integer, value varchar) WITH (partitioning = ARRAY['value'])");
```

Comment:
Partitioning on "value" is not natural. 
Also it's more natural for partitioning column to come first in the table schema: `pk integer, value varchar) ...`